site/content/docs/getting-further/script-security.md

```diff
@@ -0,0 +1,84 @@
+type: page
+tags: 
+- scripts
+EditableContent: true
+addBreadcrumbs: true
+---
+
+ZAP has as very powerful scripting capability, and supports the following scripting languages:
+* [JavaScript](/docs/desktop/addons/graalvm-javascript/) - included by default
+* [Zest](/docs/desktop/addons/zest/) - included by default
+* [BeanShell](/docs/desktop/addons/bean-shell/) - optional
+* [Kotlin](/docs/desktop/addons/kotlin-support/) - optional
+* [Python](/docs/desktop/addons/python-scripting/) - optional
+* [Ruby](/docs/desktop/addons/ruby-scripting/) - optional
+
+For more details on ZAP’s security posture see the 
+[FAQ: What is ZAP's assurance case?](/faq/what-is-zaps-assurance-case/)
+
+### Script Capabilities
+As noted on the [Script Console](/docs/desktop/addons/script-console/) page:
+
+> [!WARNING]
+> Scripts run with the same permissions as ZAP, so do not run any scripts that you do not trust!
+
+All scripts can call other scripts and any command line tools that are accessible to them based on OS permissions.
+Scripts can access any online services unless restricted by firewalls or similar.
+Scripts can typically access the Java runtime which gives them the same access to the filestore as ZAP.
+
+The one exception is Zest - this cannot run command line tools except via other scripts and cannot directly access arbitrary JVM features.
+
+### ZAP Scripts
+All of the script add-ons define templates in the relevant languages.
+
+Various ZAP add-ons and packages include scripts:
+* [Docker images](/docs/docker/about/):
+  * [httpsender](https://github.com/zaproxy/zaproxy/tree/main/docker/scripts/scripts/httpsender) scripts used to raise alerts for API issues
+* [Community Scripts](/docs/desktop/addons/community-scripts/) (optional)
+  * All of the relevant scripts in https://github.com/zaproxy/community-scripts
+* [Encoder](/docs/desktop/addons/encode-decode-hash/)
+  * Encoder JS [templates](https://github.com/zaproxy/zap-extensions/tree/main/addOns/encoder/src/main/zapHomeFiles/scripts/templates/encode-decode)
+* [Fuzzer](/docs/desktop/addons/fuzzer/)
+  * Fuzzer [templates](https://github.com/zaproxy/zap-extensions/tree/main/addOns/fuzz/src/main/zapHomeFiles/scripts/templates)
+* [OAST](/docs/desktop/addons/oast-support/) (optional)
+  * OAST related [scripts](https://github.com/zaproxy/zap-extensions/tree/main/addOns/oast/src/main/zapHomeFiles/scripts)
+* [Reports](/docs/desktop/addons/report-generation/)
+  * Scripts related to the [high-level-report](https://github.com/zaproxy/zap-extensions/tree/main/addOns/reports/src/main/zapHomeFiles/reports/high-level-report/resources)
+* [Selenium](/docs/desktop/addons/selenium/)
+  * Selenium [templates](https://github.com/zaproxy/zap-extensions/tree/main/addOns/selenium/src/main/zapHomeFiles/scripts/templates/selenium)
+* [Websockets](/docs/desktop/addons/websockets/)
+  * Websocket [templates](https://github.com/zaproxy/zap-extensions/tree/main/addOns/websocket/src/main/zapHomeFiles/scripts/templates)
```

Comment:
Should this mention the bundled passive rules? (Sorry I didn't think of it earlier)

Reply:
Those include the bundled passive scan rules, dont they?

Reply:
Well I just meant that they're not just empty or base templates, they're actual scripts with content for use.

Reply:
Yup